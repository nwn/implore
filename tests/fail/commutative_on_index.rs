use implore::impl_op;

#[derive(Copy, Clone, PartialEq, Eq, Debug)]
struct X;

#[impl_op(commutative)]
fn index(lhs: &X, idx: usize) -> &X {
    lhs
}

fn main() {}
