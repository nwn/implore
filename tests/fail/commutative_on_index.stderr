error: operation `Index` cannot commute
 --> $DIR/commutative_on_index.rs:6:11
  |
6 | #[impl_op(commutative)]
  |           ^^^^^^^^^^^
